solr/core/src/java/org/apache/solr/handler/admin/api/SnapshotAPI.java

```diff
@@ -0,0 +1,218 @@
+  @PermissionName(CORE_READ_PERM)
+  public ListSnapshotsResponse listSnapshots(
+      @Parameter(
+              description = "The name of the core for which to retrieve snapshots.",
+              required = true)
+          @PathParam("coreName")
+          String coreName) {
+    final ListSnapshotsResponse response = instantiateJerseyResponse(ListSnapshotsResponse.class);
+
+    try (SolrCore core = coreContainer.getCore(coreName)) {
+      if (core == null) {
+        throw new SolrException(
+            SolrException.ErrorCode.BAD_REQUEST, "Unable to locate core " + coreName);
+      }
+
+      SolrSnapshotMetaDataManager mgr = core.getSnapshotMetaDataManager();
+
+      final Map<String, SnapshotInformation> result = new HashMap<>();
+      for (String name : mgr.listSnapshots()) {
+        Optional<SolrSnapshotMetaDataManager.SnapshotMetaData> metadata =
+            mgr.getSnapshotMetaData(name);
+        if (metadata.isPresent()) {
+          final SnapshotInformation snapshotInformation =
+              new SnapshotInformation(
+                  metadata.get().getGenerationNumber(), metadata.get().getIndexDirPath());
+          result.put(name, snapshotInformation);
+        }
+      }
+
+      response.snapshots = result;
+    }
+
+    return response;
+  }
+
+  public static class ListSnapshotsResponse extends SolrJerseyResponse {
+    @JsonProperty(SolrSnapshotManager.SNAPSHOTS_INFO)
+    public Map<String, SnapshotInformation> snapshots;
+  }
+
+  public static class SnapshotInformation {
+    @JsonProperty(SolrSnapshotManager.GENERATION_NUM)
+    public final long generationNumber;
+
+    @JsonProperty(SolrSnapshotManager.INDEX_DIR_PATH)
+    public final String indexDirPath;
+
+    public SnapshotInformation(long generationNumber, String indexDirPath) {
+      this.generationNumber = generationNumber;
+      this.indexDirPath = indexDirPath;
+    }
+  }
+
+  @DELETE
+  @Path("/{snapshotName}")
+  @Produces({"application/json", "application/xml", BINARY_CONTENT_TYPE_V2})
+  @PermissionName(CORE_EDIT_PERM)
+  public DeleteSnapshotResponse deleteSnapshot(
+      @Parameter(
+              description = "The name of the core for which to delete a snapshot.",
+              required = true)
+          @PathParam("coreName")
+          String coreName,
+      @Parameter(description = "The name of the core snapshot to delete.", required = true)
+          @PathParam("snapshotName")
+          String snapshotName)
+      throws Exception {
+    final DeleteSnapshotResponse response = instantiateJerseyResponse(DeleteSnapshotResponse.class);
+
+    final SolrCore core = coreContainer.getCore(coreName);
+    if (core == null) {
+      throw new SolrException(
+          SolrException.ErrorCode.BAD_REQUEST, "Unable to locate core " + coreName);
+    }
+
+    try {
+      core.deleteNamedSnapshot(snapshotName);
+      // Ideally we shouldn't need this. This is added since the RPC logic in
```

Comment:
[0] I know this comment was there in the original code that you moved from DeleteSnapshotOp, so this isn't a comment on your PR, but it makes no sense to me.

What is the "this" being referred to?  Adding the coreName to the response?

Anyway, just talking out loud I guess....